src/test/scala/spray/json/AdditionalFormatsSpec.scala

```diff
@@ -64,8 +64,18 @@ class AdditionalFormatsSpec extends Specification {
   "The lazyFormat wrapper" should {
     "enable recursive format definitions" in {
       import FooProtocol._
-      Foo(1, "a", Some(Foo(2, "b", Some(Foo(3, "c") :: Nil)) :: Foo(4, "d") :: Nil)).toJson.toString mustEqual
-        """{"id":1,"name":"a","foos":[{"id":2,"name":"b","foos":[{"id":3,"name":"c"}]},{"id":4,"name":"d"}]}"""
+      val json = Foo(1, "a", Some(Foo(2, "b", Some(Foo(3, "c") :: Nil)) :: Foo(4, "d") :: Nil)).toJson.toString
+
+      val v = scala.util.Properties.versionNumberString
+      // TODO this test depeneds on internal implementations of scala.Map
+      // https://github.com/scala/scala/commit/6a570b6f1f59222cae4f
+      if (v != "2.13.0-M4") {
```

Comment:
That's pretty nasty... perhaps we could test this by parsing the formatted JSON string and checking it's equivalent to the original?

Reply:
+1